modules/v2x/v2x_proxy/app/v2x_proxy.cc

```diff
@@ -13,147 +13,201 @@
+    : node_(::apollo::cyber::CreateNode("v2x_proxy")) {
   if (node_ == nullptr) {
     AFATAL << "Create v2x proxy node failed.";
     exit(1);
   }
-
-  auto x2v_trafficlight_timer_period = ceil(
-      (1.0 / static_cast<int>(FLAGS_x2v_trafficlight_timer_frequency)) * 1000);
-  x2v_trafficlight_timer_.reset(
-      new cyber::Timer(static_cast<uint32_t>(x2v_trafficlight_timer_period),
-                       [this]() { this->OnX2vTrafficLightTimer(); }, false));
-  auto v2x_carstatus_timer_period = ceil(
-      (1.0 / static_cast<int>(FLAGS_v2x_carstatus_timer_frequency)) * 1000);
-  v2x_carstatus_timer_.reset(
-      new cyber::Timer(static_cast<uint32_t>(v2x_carstatus_timer_period),
-                       [this]() { this->OnV2xCarStatusTimer(); }, false));
-
+  internal_ = std::make_shared<InternalData>();
+  hdmap_ = std::make_shared<::apollo::hdmap::HDMap>();
+  const auto hdmap_file = apollo::hdmap::BaseMapFile();
+  if (0 != hdmap_->LoadMapFromFile(hdmap_file)) {
+    AERROR << "Failed to load hadmap file: " << hdmap_file;
+    return;
+  }
+  auto v2x_car_status_timer_period = std::ceil(
+      (1.0 / static_cast<int>(FLAGS_v2x_car_status_timer_frequency)) * 1000);
+  ::apollo::cyber::TimerOption v2x_car_status_timer_option;
+  v2x_car_status_timer_option.period =
+      static_cast<uint32_t>(v2x_car_status_timer_period);
+  v2x_car_status_timer_option.callback = [this]() {
+    this->OnV2xCarStatusTimer();
+  };
+  v2x_car_status_timer_option.oneshot = false;
+  v2x_car_status_timer_.reset(
+      new ::apollo::cyber::Timer(v2x_car_status_timer_option));
   os_interface_.reset(new OsInterFace());
   obu_interface_.reset(new ObuInterFaceGrpcImpl());
+  recv_thread_.reset(new std::thread([this]() {
+    while (!exit_) {
+      this->RecvTrafficlight();
+    }
+  }));
+  CHECK(!!hdmap_ && !!v2x_car_status_timer_ && !!os_interface_ &&
+        !!obu_interface_ && os_interface_->InitFlag() &&
+        obu_interface_->InitFlag());
+  planning_thread_.reset(new std::thread([this]() {
+    while (!exit_) {
+      this->RecvOsPlanning();
+    }
+  }));
+  v2x_car_status_timer_->Start();
+  GetRsulistFromFile(FLAGS_rsu_whitelist_name, &rsu_list_);
+  init_flag_ = true;
+}
 
-  x2v_trafficlight_ = std::make_shared<IntersectionTrafficLightData>();
-  v2x_carstatus_ = std::make_shared<CarStatus>();
-
-  if (!x2v_trafficlight_timer_ || !v2x_carstatus_timer_ || !os_interface_ ||
-      !obu_interface_ || !x2v_trafficlight_ || !v2x_carstatus_) {
-    AFATAL << "Create timer or interface failed.";
-    exit(1);
-  }
-
-  hdmap_.reset(new apollo::hdmap::HDMap());
-  std::string map_name = FLAGS_map_dir + "/" + FLAGS_base_map_filename;
-  if (FLAGS_debug_flag) {
-    map_name = FLAGS_hdmap_file_name;
-  }
-  if (hdmap_->LoadMapFromFile(map_name) != 0) {
-    AFATAL << "Failed to load hadmap file: " << FLAGS_hdmap_file_name;
-    return;
+bool V2xProxy::GetRsulistFromFile(const std::string &filename,
```

Comment:
GetRsuList

Reply:
Keep the name. Maybe add a function named ```GetRsuListFromCloud``` in next version.

Reply:
> GetRsuList

I mean the Upper/Lower. `GetRsuListFromFile`

Reply:
I mean `GetRsuListFromFile`